### PR TITLE
Feature/transcripts autoupload

### PR DESCRIPTION
**Pull request by OPersian**

### CODE REVIEW

File: video_xblock/backends/youtube.py [hunk 1]
```diff
@@ -60,3 +80,84 @@ def get_frag(self, **context):
         ))
 
         return frag
+
+    def fetch_default_transcripts_languages(self, video_id):
+        utf8_parser = etree.XMLParser(encoding='utf-8')
```
Comment (dorosh) [inline video_xblock/backends/youtube.py]: add docstring

File: video_xblock/static/html/studio_edit.html [hunk 2]
```diff
@@ -136,7 +136,83 @@
                   <span class="icon fa fa-plus" aria-hidden="true"></span>{% trans 'Add' %}<span class="sr"></span>
                 </a>
               </div>
-                <!-- End of Raccoongang addons -->
```
Comment (z4y4ts) [inline video_xblock/static/html/studio_edit.html]: Please revert

File: video_xblock/video_xblock.py [hunk 3]
```diff
@@ -544,4 +612,52 @@ def download_transcript(self, request, suffix=''):  # pylint: disable=unused-arg
             ('Content-Disposition', 'attachment; filename={}'.format(filename))
         ]
         response.headerlist = headerlist
+        return responsecurrent_time = Integer
```
Comment (z4y4ts) [inline video_xblock/video_xblock.py]: Something's wrong here

File: video_xblock/backends/base.py [hunk 4]
```diff
@@ -152,3 +182,84 @@ def add_js_content(self, path, **context):
...
         return '<script>' + self.render_resource(path, **context) + '</script>'
+
+    @abc.abstractmethod
+    def get_default_transcripts(self, **kwargs):  # pylint: disable=unused-argument
+        """
+        Fetches transcripts list from a video platform.
+
+        Arguments:
+            video_id (str): media id fetched from href field of studio-edit modal.
+        Arguments:
```
Comment (z4y4ts) [inline video_xblock/backends/base.py]: Please remove duplicate

**Comment by OPersian** [comment]
Tests will be added in the next PR. @z4y4ts please review

File: video_xblock/video_xblock.py [hunk 5]
```diff
@@ -358,17 +388,51 @@ def studio_view(self, context):  # pylint: disable=unused-argument
...
         languages = [{'label': label, 'code': lang} for lang, label in ALL_LANGUAGES]
         languages.sort(key=lambda l: l['label'])
         transcripts = json.loads(self.transcripts) if self.transcripts else []
         download_transcript_handler_url = self.runtime.handler_url(self, 'download_transcript')
+
+        # Authenticate to API of the player video platform and update metadata with auth information.
+        # Note that there is no need to authenticate to Youtube API,
+        # whilst for Wistia, a sample authorised request is to be made to ensure authentication succeeded,
+        # since it is needed for the auth status message generation and the player's state update with auth status.
+        auth_data, auth_error_message = self.authenticate_video_api()  # pylint: disable=unused-variable
```
Comment (OPersian) [inline video_xblock/video_xblock.py]: Authenticating here, we do it once for all the authorised API calls.

**Pull request merged**